src/RA_AchievementOverlay.cpp

```diff
@@ -1123,192 +1078,188 @@ void AchievementOverlay::Render(HDC hRealDC, RECT* rcDest) const
+                DrawAchievementExaminePage(hDC, nDX, nDY, rcTarget);
+                break;
 
-        case OP_LEADERBOARDS:
-            DrawLeaderboardPage(hDC, nDX, nDY, rcTarget);
-            break;
+            case Page::News:
+                DrawNewsPage(hDC, nDX, nDY, rcTarget);
+                break;
 
-        case OP_LEADERBOARD_EXAMINE:
-            DrawLeaderboardExaminePage(hDC, nDX, nDY, rcTarget);
-            break;
+            case Page::Leaderboards:
+                DrawLeaderboardPage(hDC, nDX, nDY, rcTarget);
+                break;
 
-        default:
-            //	Not implemented!
-            ASSERT(!"Attempting to render an undefined overlay page!");
-            break;
-    }
+            case Page::Leaderboard_Examine:
+                DrawLeaderboardExaminePage(hDC, nDX, nDY, rcTarget);
+                break;
 
+            default:
+                //	Not implemented!
+                ASSERT(!"Attempting to render an undefined overlay page!");
+        }
+    
     //	Title:
-    SelectObject(hDC, g_hFontTitle);
+    SelectFont(hDC, g_hFontTitle);
     SetTextColor(hDC, COL_TEXT);
-    sprintf_s(buffer, 1024, PAGE_TITLES[nCurrentPage]);
-    //sprintf_s( buffer, 1024, PAGE_TITLES[ nCurrentPage ], (*pnScrollOffset)+1 );
-    TextOut(hDC,
-        nDX + nBorder,
-        4 + nBorder,
-        NativeStr(buffer).c_str(), strlen(buffer));
 
+    const auto& sTitle{ ra::PAGE_TITLES.at(ra::etoi(nCurrentPage)) };
+    TextOut(hDC, nDX + nBorder, 4 + nBorder, sTitle, ra::narrow_cast<int>(ra::tstrlen(sTitle)));
 
-    //int nNextPage = (int)(m_nCurrentPage+1);
-    //if( nNextPage == (int)OP__MAX )
-    //	nNextPage = (int)OP_ACHIEVEMENTS;
-// 	sprintf_s( buffer, 1024, " A:%s", g_sPageTitles[nNextPage] );
-// 	TextOut( hDC, nDX+8, nHeight-24, buffer, strlen( buffer ) );
+    }
 
     //	Render controls:
-    SelectObject(hDC, g_hFontDesc2);
+    SelectFont(hDC, g_hFontDesc2);
     {
         const int nControlsX1 = 80 + 80 + 4;
         const int nControlsX2 = 80;
         const int nControlsY1 = rcTarget.bottom - 30 - 30 - 4;
         const int nControlsY2 = rcTarget.bottom - 30 - 4;
 
         //	Fill again:
-        SetRect(&rc, nRightPx - nControlsX1 - 4, nControlsY1 - 4, nRightPx, nHeight);
+        SetRect(&rc, nRightPx - nControlsX1 - 4, nControlsY1 - 4, nRightPx, lHeight);
         FillRect(hDC, &rc, g_hBrushBG);
 
         //	Draw control text:
-        sprintf_s(buffer, 1024, " ->:%s ", "Next");
-        TextOut(hDC, nRightPx - nControlsX1, nControlsY1, NativeStr(buffer).c_str(), strlen(buffer));
-
-        sprintf_s(buffer, 1024, " <-:%s ", "Prev");
-        TextOut(hDC, nRightPx - nControlsX1, nControlsY2, NativeStr(buffer).c_str(), strlen(buffer));
+        {
+            ra::tstring stNext{ _T(" ->:") };
+            stNext += _T("Next ");
+            TextOut(hDC, nRightPx - nControlsX1, nControlsY1, stNext.c_str(), ra::narrow_cast<int>(stNext.length()));
+        }
```

Comment:
An example of where `ra::narrow_cast` was used. Boxing happens internally so we didn't have to do it here.